### ISSUE TITLE
Question about cmake Tpetra detection

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#14372]
We have the following piece of code: https://github.com/dealii/dealii/blob/master/cmake/configure/configure_20_trilinos.cmake#L230-L254 Specifically, we have this code snippet we want to compile:
```
          using LO       = int;
          using GO       = unsigned int;
          using GO       = ${_global_index_type};
```
It is, interestingly, allowed to *redeclare* the same type via `using`, but only if the right hand side is the same. It is *not* allowed to declare it to a different type. In other words, compiling this code snippet will only work if `${_global_index_type}` is `unsigned int`.

Is this a mistake in our cmake scripts? Or am I just not getting what this is trying to achieve?

**Comment by drwells** [comment]
Possibly related to #13972.

**Comment by tamiko** [comment]
@bangerth The line should not be repeated - this is a refactoring mistake.

Unfortunately, we have not resolved #13972 yet, meaning this configure check will always fail for any (default configured) Trilinos version after 13.2 because the Tpetra/Kokkos backend need *signed* ordinals.
